github/github.go

```diff
@@ -934,8 +936,8 @@ func (c *Client) bareDo(ctx context.Context, caller *http.Client, req *http.Requ
 			err = aerr
 		}
 
-		rateLimitError, ok := err.(*RateLimitError)
-		if ok && req.Context().Value(SleepUntilPrimaryRateLimitResetWhenRateLimited) != nil {
+		var rateLimitError *RateLimitError
+		if ok := errors.As(err, &rateLimitError); ok && req.Context().Value(SleepUntilPrimaryRateLimitResetWhenRateLimited) != nil {
```

Comment:
Everywhere you see this pattern, please change it to the simplified version:


```suggestion
		if errors.As(err, &rateLimitError) && req.Context().Value(SleepUntilPrimaryRateLimitResetWhenRateLimited) != nil {
```

Reply:
Oh, this is hard to find! Thanks!